packages/spacecat-shared-google-client/src/index.js

```diff
@@ -148,6 +149,34 @@ export default class GoogleClient {
     }
   }
 
+  async urlInspect(url) {
+    if (!isValidUrl(url)) {
+      throw new Error('Error inspecting URL: Invalid URL format');
+    }
+
+    await this.#refreshTokenIfExpired();
+
+    const apiEndpoint = 'https://searchconsole.googleapis.com/v1/urlInspection/index:inspect';
+
+    const response = await httpFetch(apiEndpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${this.authClient.credentials.access_token}`,
+      },
+      body: JSON.stringify({
+        inspectionUrl: url,
+        siteUrl: this.siteUrl,
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error('Error inspecting URL');
```

Comment:
It may be helpful logging which url had an error while inspecting and what status was returned.